packages/apps/tools/src/pages/modules/explorer/utils.ts

```diff
@@ -48,3 +50,93 @@ export function getCookieValue<Expected>(
   }
   return defaultValue ?? null;
 }
+
+export type Namespace = string;
+export type ModuleName = string;
+export type ModuleData = Array<IncompleteModuleModel>;
+
+export type ModulesMap = Map<
+  Namespace,
+  ModuleData | Map<ModuleName, ModuleData>
+>;
+
+export const mapToTreeItems = (
+  modulesMap: Map<Namespace, ModuleData | Map<ModuleName, ModuleData>>,
+  parent?: Namespace,
+): TreeItem<IncompleteModuleModel>[] => {
+  return [...modulesMap]
+    .sort(([nameA], [nameB]) => nameA.localeCompare(nameB))
+    .map(([name, value]) => {
+      let children;
+
+      if (value instanceof Map) {
+        const mapped = mapToTreeItems(
+          value,
+          `${typeof parent === 'string' ? `${parent}.` : ''}${name}`,
+        );
+        children = mapped;
+      } else {
+        children = value.map((chain) => {
+          return {
+            data: chain,
+            key: `${typeof parent === 'string' ? `${parent}.` : ''}${name}.${chain.chainId}`,
+            title: chain.chainId,
+            children: [],
+            label: chain.hash,
+          };
+        });
+      }
+
+      // Let's take the info from the first child as our data
+      const aux: IncompleteModuleModel = {
+        name: children[0].data.name,
+        chainId: children[0].data.chainId,
+        networkId: children[0].data.networkId,
+      };
+
+      return {
+        data: aux,
+        key: `${typeof parent === 'string' ? `${parent}.` : ''}${name}`,
+        title: name,
+        children,
+      };
+    });
+};
+
+export const modelsToTreeMap = (
+  models: Array<IncompleteModuleModel>,
+): ModulesMap => {
+  const modulesMap: ModulesMap = new Map();
+
+  models.forEach((module) => {
+    const [namespace, moduleName] = module.name.split('.');
+
+    if (moduleName) {
+      if (!modulesMap.has(namespace)) {
+        modulesMap.set(namespace, new Map());
+      }
+
+      const map = modulesMap.get(namespace)! as Map<ModuleName, ModuleData>;
+
+      if (!map.has(moduleName)) {
+        map.set(moduleName, []);
+      }
+
+      const moduleList = map.get(moduleName)!;
+
+      moduleList.push(module);
+    } else {
+      if (!modulesMap.has(namespace)) {
+        modulesMap.set(namespace, []);
+      }
+
+      const moduleList = modulesMap.get(
+        namespace,
+      )! as Array<IncompleteModuleModel>;
+
+      moduleList.push(module);
+    }
+  });
+
+  return modulesMap;
+};
```

Comment:
Do you have tests for this file? It quite hard to read what this is about if you don't know the context. Tests might help and a few comments would be even better.

Reply:
Agreed, will do!

Reply:
Tests and some documentation added

Reply:
Very nice!